### ISSUE TITLE
gui: Gtk 3.20 breaks the progressBar

### ISSUE THREAD

**Issue by 12345ieee** [scanmem/scanmem#167]
3.20 is now the default Gtk version on Debian testing, so I updated.
Clearly this broke a piece of GC.

Specifically the progressBar is stuck at 0 and jumps at 1 only at the end, making it utterly useless.

The problem is here: https://github.com/scanmem/scanmem/blob/master/gui/GameConqueror.py#L779
the `Gdk.flush()` used to redraw the bar, now it doesn't do it any more (the bar internal progress is correctly set).

Digging more, this seems to be an issue of threading.
The helpful reference [here](https://developer.gnome.org/gdk3/stable/gdk3-Threads.html#gdk-threads-enter) tells us threads were deprecated 14 versions ago.

-----------

If I can give my opinion, the more I look into GC, the more I feel the graphical part needs a rewrite, 5 years changed the Gtk quite a lot and dealing with 10 legacy lines breaking for every version has been extremely annoying.

**Comment by 12345ieee** [comment]
I was wrong, the culprit is
https://github.com/scanmem/scanmem/blob/master/gui/GameConqueror.py#L895
commenting it, the problem disappears.

Hopefully the documentation will tell me how to fix this while keeping the main window freeze.

**Comment by sriemer** [comment]
Sounds like the progress bar is also disabled and not updated this way.
What about trying the enable it again or to disable everything but the progress bar (cumbersome!)?

**Comment by 12345ieee** [comment]
If you enable only the pbar, you get nothing, as you need the whole chain of parents to be enabled up to the window itself.

Barring some reshuffling of the gui, to put every other thing into 3-4 easy-to-disable containers, the only way is to manually disable every little element, which is very very cumbersome.

**Comment by 12345ieee** [comment]
Ok, after some work reshuffling the containers, this issue is fixed in my gui branch:
https://github.com/scanmem/scanmem/compare/master...12345ieee:gui

The most reasonable way I found was to reorganize the whole container hierarchy, to have widgets grouped in grids, so that I could disable them in groups.
Grouping widgets has other advantages, like not getting mad aligning 10 buttons or not having the layout broken again when gtk 3.22 hits.

That branch is in a terrible state and has a huge amount of completely unrelated changes (#168 , removed all deprecated widgets, bunch of random fixes...) next week-end I'll probably get around to squash, edit and submit it as 2-3 separate pull requests.

TL; DR: Don't work on this, it's done and will be submitted in a couple weeks.

**Comment by sriemer** [comment]
Nice! Thanks! Waiting for your pull requests then.
